service/frontend/wrappers/clusterredirection/policy.go

```diff
@@ -340,3 +324,40 @@ func (policy *selectedOrAllAPIsForwardingRedirectionPolicy) getTargetClusterAndI
 
 	return currentActiveCluster, true
 }
+
+func (policy *selectedOrAllAPIsForwardingRedirectionPolicy) activeClusterForActiveActiveDomainRequest(
+	ctx context.Context,
+	domainEntry *cache.DomainCacheEntry,
+	workflowExecution *types.WorkflowExecution,
+	actClSelPolicyForNewWF *types.ActiveClusterSelectionPolicy,
+	apiName string,
+) string {
+	policy.logger.Debug("Determining active cluster for active-active domain request", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.Dynamic("execution", workflowExecution), tag.OperationName(apiName))
+	if actClSelPolicyForNewWF != nil {
+		policy.logger.Debug("Active cluster selection policy for new workflow", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.OperationName(apiName), tag.Dynamic("policy", actClSelPolicyForNewWF))
+		lookupRes, err := policy.activeClusterManager.LookupNewWorkflow(ctx, domainEntry.GetInfo().ID, actClSelPolicyForNewWF)
+		if err != nil {
+			policy.logger.Error("Failed to lookup active cluster of new workflow, using current cluster", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.OperationName(apiName), tag.Error(err))
+			return policy.currentClusterName
+		}
+		return lookupRes.ClusterName
+	}
+
+	if workflowExecution == nil {
+		policy.logger.Debug("Workflow execution is nil, using current cluster", tag.WorkflowDomainName(domainEntry.GetInfo().Name), tag.OperationName(apiName))
+		return policy.currentClusterName
+	}
+	if workflowExecution.RunID == "" {
```

Comment:
what happens if it's a signal/cancel/terminate request without runID?

Reply:
Those will be forwarded to history in current cluster and it will return a DomainNotActiveError. I have a todo make sure DomainNotActiveError is populated with the appropriate active cluster name. I will make a small change in this PR to avoid subsequent forwarding to active cluster in DomainNotActiveError for active-active domains. thanks for calling this out